bin/ch_misc.c

```diff
@@ -640,3 +666,37 @@ void version(void)
 {
    fprintf(stderr, "%s\n", VERSION);
 }
+
+/* Append null-terminated string “str” to the memory buffer “offset” bytes away
+   from the address pointed to by “addr”. Buffer should be “size” bytes long.
+   Return the number of bytes written. If there isn’t enough room for the
+   string, do nothing and return zero. */
+size_t warnings_append(char *addr, char *str, size_t size, size_t offset)
```

Comment:
This is pretty generic, yes? More about strings than warnings? If so name it accordingly; if not, explain why it's not generic.